taskfiles/charts.yml

```diff
@@ -0,0 +1,129 @@
+      - nix develop -c staticcheck ./charts/...
+
+  generate:
+    desc: "Run all file generation tasks"
+    cmds:
+      # Generate chart README.md's
+      - nix develop -c helm-docs -c ./charts/
+      - task: generate:console
+      - task: generate:operator
+      - task: generate:redpanda
+      - task: generate:connectors
+      # Ensure go deps are up to date
+      - task: go-deps:charts
+      - task: go-deps:genpartial
+      - task: go-deps:genschema
+      - task: go-deps:gotohelm
+      - task: go-deps:pkg
+      # Ensure go deps in the gotohelm test package are up to date
+      - cd ./pkg/gotohelm/testdata/src/example && go mod tidy
+      - go work sync
+      # Ensure flake.nix has been formatted.
+      - nix fmt
+      - task: :dev:update-licenses
+
+  generate:console:
+    desc: "Generate files for the console Helm chart"
+    cmds:
+      # Generate a "partial" version of the Values struct.
+      - go run ./genpartial -out charts/console/values_partial.gen.go -struct Values ./charts/console
+      # Generate helm templates from Go definitions
+      - go run ./gotohelm -write ./charts/console/templates ./charts/console
+
+  generate:operator:
+    desc: "Generate files for the operator Helm chart"
+    cmds:
+      # Generate a "partial" version of the Values struct.
+      - go run ./genpartial -out charts/operator/values_partial.gen.go -struct Values ./charts/operator
+      # Generate the values JSON schema from the Values struct
+      - go run ./genschema operator > charts/operator/values.schema.json
+      # Generate helm templates from Go definitions
+      - go run ./gotohelm -write ./charts/operator/templates ./charts/operator
+
+  generate:redpanda:
+    desc: "Generate files for the redpanda Helm chart"
+    cmds:
+      # Generate a "partial" version of the Values struct.
+      - go run ./genpartial -out charts/redpanda/values_partial.gen.go -struct Values ./charts/redpanda
+      # Generate the values JSON schema from the Values struct
+      - go run ./genschema redpanda > charts/redpanda/values.schema.json
+      # Generate helm templates from Go definitions
+      - go run ./gotohelm -write ./charts/redpanda/templates ./charts/redpanda ./charts/...
+
+  generate:connectors:
+    desc: "Generate files for the connectors Helm chart"
+    cmds:
+      # Generate a "partial" version of the Values struct.
+      - go run ./genpartial -out charts/connectors/values_partial.gen.go -struct Values ./charts/connectors
+      # Generate helm templates from Go definitions
+      - go run ./gotohelm -write ./charts/connectors/templates ./charts/connectors
+
+  go-deps:charts:
+    dir: 'charts'
+    cmds:
+      - go mod tidy
+  go-deps:genpartial:
+    dir: 'charts'
+    cmds:
+      - go mod tidy
+  go-deps:genschema:
+    dir: 'charts'
+    cmds:
+      - go mod tidy
+  go-deps:gotohelm:
+    dir: 'charts'
+    cmds:
+      - go mod tidy
+  go-deps:pkg:
+    dir: 'pkg'
+    cmds:
+      - go mod tidy
+
+  tests:
```

Comment:
Similarly, it'd be preferable to have this at a top level `test`. We could do something like:

```yaml
test:
  var:
    PKG: "./..."
  cmds:
  - go test {{CLI_FLAGS}} {{.PKG}}
``` 

In CI you could then invoke it as:
```bash
task test PKG=./charts/... -- -v -coverprofile cover.out -timeout 20m
```

Reply:
Ok, but I would like to use `gotestsum` to generate junit xml, so that it's recognised by the buildkite. I'm already annoyed by the instability of our CI. Every time I need to check which test this time failed can be replaced with simple "oh at the top I see that test failed".